cns/restserver/api_test.go

```diff
@@ -515,25 +515,60 @@ func createNC(
 		t.Fatal(err)
 	}
 
-	publishNCViaCNS(t, params.vnetID, params.ncID)
+	createNetworkContainerURL := "http://" + nmagentEndpoint +
+		"/machine/plugins/?comp=nmagent&type=NetworkManagement/interfaces/dummyIntf/networkContainers/dummyNCURL/authenticationToken/dummyT/api-version/"
+
+	err := publishNCViaCNS(t, params.vnetID, params.ncID, createNetworkContainerURL, false)
+	if err != nil {
+		t.Fatal(err)
+	}
 }
 
 func TestPublishNCViaCNS(t *testing.T) {
 	fmt.Println("Test: publishNetworkContainer")
-	publishNCViaCNS(t, "vnet1", "ethWebApp")
+
+	createNetworkContainerURL := "http://" + nmagentEndpoint +
+		"/machine/plugins/?comp=nmagent&type=NetworkManagement/interfaces/dummyIntf/networkContainers/dummyNCURL/authenticationToken/dummyT/api-version/1"
+	err := publishNCViaCNS(t, "vnet1", "ethWebApp", createNetworkContainerURL, false)
+	if err != nil {
+		t.Fatal(err)
```

Comment:
more context on logged error. same for all below

Reply:
Added